app/notification/application/map_contents.py

```diff
@@ -34,15 +35,16 @@ def from_json(json_data: dict):
             ApplicationData object with application contents.
         """
         data = Notification.from_json(json_data)
-        application = data.content["application"]
-        return Application(
-            contact_info=data.contact_info,
-            questions=Application.process_questions_and_answers(data),
-            submission_date=application.get("date_submitted"),
-            fund_name=application.get("project_name"),
-            fund_round=application.get("round_id"),
-            application_id=application.get("id"),
-        )
+        if data.template_type == "APPLICATION_RECORD_OF_SUBMISSION":
```

Comment:
Do we need this `if` here? Won't it only have got to this section if that's the template type? Also what happens if that's false?

Reply:
I pushed some changes instead of stashing. The PR was mainly about application-conformation email with file attachment but I was doing some other study while waiting for your merge so please  ignore it. I will be refactoring them as discussed